docarray/array/storage/sqlite/getsetdel.py

```diff
@@ -52,20 +52,28 @@ def _del_docs_by_ids(self, ids: str) -> Iterable['Document']:
         self._commit()
 
     def _load_offset2ids(self):
-        r = self._sql(
-            f"SELECT doc_id FROM {self._table_name} ORDER BY item_order",
-        )
-        self._offset2ids = Offset2ID(list(map(itemgetter(0), r)))
+        if self._list_like:
+            r = self._sql(
+                f"SELECT doc_id FROM {self._table_name} ORDER BY item_order",
+            )
+            self._offset2ids = Offset2ID(
+                list(map(itemgetter(0), r)), list_like=self._list_like
+            )
+            return self._offset2ids
```

Comment:
keep only a single return outside the if

Reply:
actually, why are we returning?

Reply:
That's what @alaeddine-13 suggests to return something, and when the flag is disabled, just return empty. Considered there is no current usage outside the method, I just return the offset2id.